src/storage/mvcc/reader/reader.rs

```diff
@@ -474,6 +475,44 @@ impl<S: Snapshot> MvccReader<S> {
     }
 }
 
+pub fn extract_mvcc_props(collection: TablePropertiesCollection) -> Option<MvccProperties> {
+    if collection.is_empty() {
+        return None;
+    }
+    let mut props = MvccProperties::new();
+    for (_, v) in &*collection {
+        let mvcc = match MvccProperties::decode(v.user_collected_properties()) {
+            Ok(v) => v,
+            Err(_) => return None,
+        };
+        props.add(&mvcc);
+    }
+    Some(props)
+}
+
+pub fn need_gc(props: MvccProperties, safe_point: u64, ratio_threshold: f64) -> bool {
+    if ratio_threshold <= 1.0 {
```

Comment:
This seems to be duplicated with `Reader::need_gc`. I think you can invoke this function in `Reader::need_gc`.